### ISSUE TITLE
Cannot update plugins - XrmToolBox v1.2017.7.18

### ISSUE THREAD

**Issue by rajyraman** [MscrmTools/XrmToolBox#672]
Getting an error when I select the plugins to update and press the "Install" button.

![image](https://user-images.githubusercontent.com/5035266/30936962-cee00e92-a418-11e7-9134-267aaefe1066.png)

```csharp
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at System.Collections.ArrayList.ArrayListEnumeratorSimple.MoveNext()
   at System.Windows.Forms.Application.ExitInternal()
   at System.Windows.Forms.Application.Restart()
   at XrmToolBox.PluginsStore.Store.PerformInstallation(PluginUpdates updates)
   at XrmToolBox.PluginsStore.StoreForm.tsbInstall_Click(Object sender, EventArgs e)
   at System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   at System.Windows.Forms.ToolStripButton.OnClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   at System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   at System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ToolStrip.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

**Comment by rappen** [comment]
I guess these are related... https://github.com/Innofactor/XrmToolBox.PluginTraceViewer/issues/65

**Comment by medicstuder** [comment]
@rappen I can confirm that is the cause of the problem being reported.

**Comment by daryllabar** [comment]
@rappen do you have any idea what is causing this?

**Comment by rappen** [comment]
I have changed something in the constructor in this attached version - I would be very glad if you could test with this file instead and report the behaviour!
[PluginTraceViewer 1.2017.10.991.zip](https://github.com/MscrmTools/XrmToolBox/files/1354721/PluginTraceViewer.1.2017.10.991.zip)

Ping @rajyraman @medicstuder @daryllabar @peterloudon

**Comment by rajyraman** [comment]
@rappen Not happening anymore after using the new PTV assembly.

**Comment by medicstuder** [comment]
same here, not happening anymore

**Comment by rappen** [comment]
Thanks a lot guys!
Strange behavior in XTB core combined with not-very-optimal code in new PTV. Unfortunately not reproducible everywhere.
Will release updated PTV shortly!

**Comment by rappen** [comment]
Fix is now published in PTV version 1.2017.10.1.
